src/Analysis/GWASV2/Steps/SelectCovariates/SelectCovariates.jsx

```diff
@@ -54,12 +67,23 @@ const SelectCovariates = ({
               height: 60,
               marginLeft: 5,
             }}
-            onClick={() => setMode('dichotomous')}
+            onClick={() => setSelectionMode('dichotomous')}
           >
             Add Dichotomous Outcome Covariate
           </button>
         </div>
       )}
+      <CovariatesCardsList
+        covariates={covariates}
+        deleteDichotomousCovariate={(chosenDichotomousCovariate) => dispatch({
+          type: ACTIONS.DELETE_DICHOTOMOUS_COVARIATE,
```

Comment:
maybe we could have just one action to delete covariate instead of one action per covariate type?

Reply:
Updated to 
```
        deleteCovariate={(chosenCovariate) => dispatch({
          type: ACTIONS.DELETE_COVARIATE,
          payload: chosenCovariate,
        })}
```